fw/http_match.c

```diff
@@ -874,25 +880,47 @@ tfw_http_verify_hdr_field(tfw_http_match_fld_t field, const char **hdr_name,
  * @is_resp_hdr - header name identifier:
  *		  true for `Set-Cookie`,
  *		  false for `Cookie`.
- * @return - 0 if given cookie name hasn't been found,
- *	     1 if found + particular cookie value (if exists)
+ * @return - -ENOMEM if memory allocation fails.
+ *           0 if given cookie name hasn't been found,
+ *           count of found cookies + particular cookie values (if exists)
  */
 int
-tfw_http_search_cookie(const char *cstr, unsigned long clen,
-		       const TfwStr *cookie, TfwStr *val,
-		       tfw_http_match_op_t op, bool is_resp_hdr)
+tfw_http_search_cookie(const TfwHttpReq *req, const char *cstr,
+		       unsigned long clen, const TfwStr *cookie,
+		       TfwStr *val, tfw_http_match_op_t op,
+		       bool is_resp_hdr)
 {
 	TfwStr *chunk, *end;
 	TfwStr tmp = { 0 };
 	unsigned int n = cookie->nchunks;
+	int cnt = 0;
+
+#define TFW_COLLECT_COOKIE(req, chunk, val)				\
```

Comment:
I change this macro to TFW_COLLECT_REQ_COOKIE with only one argument req, because this marco requires a lot of argument otherwise. As i see in http_parser.c we have a lot of macros which mix directly variable usage and arguments